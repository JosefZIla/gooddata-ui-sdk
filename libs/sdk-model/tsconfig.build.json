{
    "extends": "./tsconfig.json",
    "include": ["src/index.ts", "src/dashboard/drillUrl.ts"],
    "compilerOptions": {
        "baseUrl": null,
        "paths": null,
        "declaration": true,
        "declarationMap": true
    }
}
