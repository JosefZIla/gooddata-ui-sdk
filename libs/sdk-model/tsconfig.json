{
    "compilerOptions": {
        "esModuleInterop": true,
        "target": "es5",
        "module": "commonjs",
        "moduleResolution": "node",
        "importHelpers": true,
        "lib": ["es6", "dom", "ES2016.Array.Include"],
        "declaration": true,
        "declarationMap": true,
        "sourceMap": true,
        "outDir": "./dist",
        "strict": true,
        "noImplicitAny": true,
        "noUnusedLocals": true,
        "noUnusedParameters": true,
        "noImplicitReturns": true,
        "removeComments": false
    },
    "exclude": ["**/node_modules", "dist", "esm"]
}
