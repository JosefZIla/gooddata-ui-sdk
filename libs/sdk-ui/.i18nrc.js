module.exports = {
    paths: ["./src/base/localization/bundles"],
    structure: true,
    intl: true,
    html: true,
    insightToReport: true,
    usage: false,
    debug: false,
    source: "src/**/*.{ts,js,tsx,jsx}",
    rules: [
        {
            pattern: [/.+/],
        },
    ],
};
