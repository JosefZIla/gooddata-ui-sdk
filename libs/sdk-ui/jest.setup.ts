// (C) 2019 GoodData Corporation
/* eslint-disable import/no-unassigned-import */
import * as raf from "raf";
import "@testing-library/jest-dom/extend-expect";

raf.polyfill();
