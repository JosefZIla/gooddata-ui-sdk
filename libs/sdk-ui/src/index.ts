// (C) 2007-2021 GoodData Corporation
/**
 * @packageDocumentation
 */
import "./polyfills";
// new exports

export * from "./base";
export * from "./execution";
export * from "./kpi";

/**
 * Common interface uses to specify number separators for the different SDK components.
 * @public
 */
export { ISeparators } from "@gooddata/sdk-backend-spi";
