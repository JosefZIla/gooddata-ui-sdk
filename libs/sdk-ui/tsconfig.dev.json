{
    "extends": "./tsconfig.build.esm.json",
    "compilerOptions": {
        "noImplicitAny": false,
        "noUnusedLocals": false,
        "noUnusedParameters": false,
        "pretty": true
    }
}
