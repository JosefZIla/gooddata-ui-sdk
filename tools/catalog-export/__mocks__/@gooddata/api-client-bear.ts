// (C) 2007-2018 GoodData Corporation
import catalogItemsResponse from "./loadItems.json";
import getObjectsResponse from "./getObjects.json";
import getObjectsByQueryResponse from "./getObjectsByQuery.json";
import getVisualizationsResponse from "./getVisualizations.json";
import getAnalyticalDashboards from "./getAnalyticalDashboards.json";

export default {
    catalogue: {
        loadItems() {
            return Promise.resolve(catalogItemsResponse);
        },
    },
    md: {
        getObjectsByQuery() {
            return Promise.resolve(getObjectsByQueryResponse);
        },
        getObjects() {
            return Promise.resolve(getObjectsResponse);
        },
        getVisualizations() {
            return Promise.resolve(getVisualizationsResponse);
        },
        getAnalyticalDashboards() {
            return Promise.resolve(getAnalyticalDashboards);
        },
    },
};
