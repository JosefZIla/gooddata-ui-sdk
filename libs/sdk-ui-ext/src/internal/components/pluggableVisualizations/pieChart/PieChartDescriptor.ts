// (C) 2021-2022 GoodData Corporation

import {
    IVisualizationDescriptor,
    PluggableVisualizationFactory,
} from "../../../interfaces/VisualizationDescriptor";
import { PluggablePieChart } from "./PluggablePieChart";
import { BaseChartDescriptor } from "../baseChart/BaseChartDescriptor";
import { getReactEmbeddingCodeGenerator } from "../../../utils/embeddingCodeGenerator";
import {
    bucketConversion,
    chartConfigFromInsight,
    getInsightToPropsConverter,
    insightConversion,
} from "../../../utils/embeddingCodeGenerator/insightToPropsConverter";
import { IPieChartProps } from "@gooddata/sdk-ui-charts";
import { BucketNames } from "@gooddata/sdk-ui";
import { bucketMeasures, insightFilters, insightSorts, bucketAttribute } from "@gooddata/sdk-model";

export class PieChartDescriptor extends BaseChartDescriptor implements IVisualizationDescriptor {
    public getFactory(): PluggableVisualizationFactory {
        return (params) => new PluggablePieChart(params);
    }

    public getEmbeddingCode = getReactEmbeddingCodeGenerator({
        component: {
            importType: "named",
            name: "PieChart",
            package: "@gooddata/sdk-ui-charts",
        },
        insightToProps: getInsightToPropsConverter<IPieChartProps>({
            measures: bucketConversion("measures", BucketNames.MEASURES, bucketMeasures),
            viewBy: bucketConversion("viewBy", BucketNames.VIEW, bucketAttribute),
            filters: insightConversion("filters", insightFilters),
            sortBy: insightConversion("sortBy", insightSorts),
            config: insightConversion("config", chartConfigFromInsight),
        }),
    });
}
