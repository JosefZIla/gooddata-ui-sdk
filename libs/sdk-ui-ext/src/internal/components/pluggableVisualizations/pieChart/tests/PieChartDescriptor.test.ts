// (C) 2022 GoodData Corporation
import { ReferenceMd } from "@gooddata/reference-workspace";
import { modifyAttribute, modifyMeasure, newBucket, newInsightDefinition } from "@gooddata/sdk-model";
import { BucketNames } from "@gooddata/sdk-ui";
import { PieChartDescriptor } from "../PieChartDescriptor";

describe("PieChartDescriptor", () => {
    it("should generate embedding code without context", () => {
        // TODO replace by some recording?
        const insight = newInsightDefinition("local:pie", (i) =>
            i.buckets([
                newBucket(
                    BucketNames.MEASURES,
                    modifyMeasure(ReferenceMd.Amount, (m) =>
                        m.alias("Custom Amount alias").localId("amount-custom-local-id"),
                    ),
                ),
                newBucket(
                    BucketNames.VIEW,
                    modifyAttribute(ReferenceMd.Region, (a) =>
                        a.alias("Custom Region alias").localId("region-custom-local-id"),
                    ),
                ),
            ]),
        );
        const descriptor = new PieChartDescriptor();

        expect(descriptor.getEmbeddingCode(insight)).toMatchSnapshot();
    });
});
