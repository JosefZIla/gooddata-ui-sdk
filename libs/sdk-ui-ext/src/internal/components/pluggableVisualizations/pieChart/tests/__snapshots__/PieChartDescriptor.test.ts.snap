// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PieChartDescriptor should generate embedding code without context 1`] = `
"import React from \\"react\\";
import { idRef, newAttribute, newMeasure } from \\"@gooddata/sdk-model\\";
import { PieChart } from \\"@gooddata/sdk-ui-charts\\";

const measures = [
    newMeasure(idRef(\\"aangOxLSeztu\\", \\"measure\\"), m => m.alias(\\"Custom Amount alias\\").localId(\\"m_aangOxLSeztu\\"))
];
const viewBy = newAttribute(idRef(\\"label.owner.region\\", \\"displayForm\\"), a => a.alias(\\"Custom Region alias\\").localId(\\"a_label.owner.region\\"));
const style = { height: 400 };

function MyComponent() {
    return (
        <div style={style}>
            <PieChart
                measures={measures}
                viewBy={viewBy}
            />
        </div>
    );
}
"
`;
